internal/os/smb/api.go

```diff
@@ -14,8 +15,47 @@ func New() APIImplementor {
 	return APIImplementor{}
 }
 
-func (APIImplementor) NewSmbGlobalMapping(remotePath, username, password string) error {
-	klog.V(4).Infof("NewSmbGlobalMapping: remotePath:%q", remotePath)
+func (APIImplementor) IsSmbMapped(remotePath string) (bool, error) {
+	cmdLine := fmt.Sprintf(`$(Get-SmbGlobalMapping -RemotePath $Env:smbremotepath -ErrorAction Stop).Status `)
+	cmd := exec.Command("powershell", "/c", cmdLine)
+	cmd.Env = append(os.Environ(),
+		fmt.Sprintf("smbremotepath=%s", remotePath))
+
+	out, err := cmd.CombinedOutput()
+	if err != nil {
+		return false, fmt.Errorf("error checking smb mapping %s, err: %v", remotePath, err)
+	}
+
+	if len(out) == 0 || !strings.EqualFold(strings.TrimSpace(string(out)), "OK") {
+		return false, nil
+	}
+	return true, nil
+}
+
+// SMBLink - creates a direcotry symbolic link to the remote share.
+// The os.Symlink was having issue for cases where the destination was an SMB share - the container
+// runtime would complain stating "Access Denied". Because of this, we had to perform
+// this operation with powershell commandlet creating an directory softlink.
+// Since os.Symlink is currently being used in working code paths, no attempt is made in
+// alpha to merge the paths.
+// TODO (for beta release): Merge the link paths - os.Symlink and Powershell link path.
+func (APIImplementor) SMBLink(remotePath, localPath string) error {
+	cmdLine := fmt.Sprintf(`New-Item -ItemType SymbolicLink $Env:smblocalPath -Target $Env:smbremotepath`)
+	cmd := exec.Command("powershell", "/c", cmdLine)
+	cmd.Env = append(os.Environ(),
+		fmt.Sprintf("smbremotepath=%s", remotePath),
+		fmt.Sprintf("smblocalpath=%s", localPath),
+	)
+	_, err := cmd.CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("error linking %s to %s, err: %v", remotePath, localPath, err)
+	}
+
+	return nil
+}
+
+func (APIImplementor) NewSmbGlobalMapping(remotePath, localPath, username, password string) error {
+	klog.V(4).Infof("NewSmbGlobalMapping: remotePath:%q, localPath:%q", remotePath, localPath)
```

Comment:
is there any restrictions on localpath?
Could you also comment on the use of localpath?